### ISSUE TITLE
[🐞] [vite-plugin-qwik] Found '$' but did not find the corresponding 'Qrl' exported in the same file.

### ISSUE THREAD

**Issue by motss** [BuilderIO/qwik#5206]
### Which component is affected?

Qwik Rollup / Vite plugin

## Background

I'm trying to create a monorepo using Qwik City where all the codes reside in the `src/` folder. There is a `packages` folder where it hosts all the packages to be published. Anything in the `src` are basically files and folders needed for a website which will render demos of each and every package from the `packages`.

The basic idea is to create any package you want in the `packages` and then import them into the demo.

```sh
\
|- packages
|    `- button
|     
|- src
|   |- components // button could be imported here like a installed package, e.g. import Button from '@ui/button';
|   |- routes
|   |- root.tsx
|   |- entry.ssr.tsx
...
```

```ts
// Basic code implementation of Button.tsx
export const Button = component$((props) => {
  return <button {...props}><Slot /></button>
});
```

```ts
// Basic code showing how to import Button

import { Button } from '@ui/button';

export default component$(() => {
  return (
    <Button>Click me</Button>
  );
});
```

## Problem

Everything works great when running `pnpm dev` locally. However, when running `pnpm preview`, the build will fail with the following error:

```sh
[vite-plugin-qwik] Found '$' but did not find the corresponding 'Qrl' exported in the same file. Please check that it is exported and spelled correctly
```

### Reproduction

https://stackblitz.com/edit/qwik-starter-tcqf9b?file=src%2Fpackages%2Fqwik-button%2Fpackage.json

### Steps to reproduce

1. Go to the reproduction URL
2. Terminate any running command in an active terminal then run `pnpm install && pnpm preview`
3. You should see `[vite-plugin-qwik] Found '$' but did not find the corresponding 'Qrl' exported in the same file. Please check that it is exported and spelled correctly`.

### System Info

```shell
envinfo@7.10.0

  System:
    OS: macOS 13.6
    CPU: (10) arm64 Apple M1 Max
    Memory: 25.99 GB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.7.0 - ~/Library/Caches/fnm_multishells/1812_1695370805799/bin/node
    npm: 10.1.0 - ~/Library/Caches/fnm_multishells/1812_1695370805799/bin/npm
    pnpm: 8.7.6 - /opt/homebrew/bin/pnpm
  Browsers:
    Chrome: 116.0.5845.187
    Safari: 16.6
    Safari Technology Preview: 17.0
  npmPackages:
    @builder.io/qwik: ^1.2.12 => 1.2.12 
    @builder.io/qwik-city: ^1.2.12 => 1.2.12 
    undici: 5.25.1 => 5.25.1 
    vite: 4.4.9 => 4.4.9
```

**Comment by mhevery** [comment]
In your reproduction, you have a `packages` folder inside of the `src` folder. This is not allowed. Please create a reproduction when there are no packages, only source code in the src folder, and then share it here.

If things work fine, then please close the issue.

**Comment by motss** [comment]
@mhevery Thanks for the recommendation. 👍  I relocated the `packages` from the `src` directory, but it didn't function right away. I had to include `"qwik": "./src/qwik-button.tsx"` to get it functioning correctly.

However, I discovered that Qwik does not handle `dependencies` of a component (be it in a monorepo or in library mode) that well. 🤷  It forces us devs or the library users to always install all `dependencies` to the root workspace. I will raise a separate issue for that. 

Closing this as it has been resolved with this [comment](https://github.com/BuilderIO/qwik/issues/5206#issuecomment-1736267361).
